src/core/execute.c

```diff
@@ -3012,6 +3012,33 @@ static int exec_parameters_get_cgroup_path(const ExecParameters *params, char **
         return using_subcgroup;
 }
 
+static int exec_context_cpu_affinity_from_numa(const ExecContext *c, CPUSet *ret) {
+        _cleanup_(cpu_set_reset) CPUSet s = {};
+        int r;
+
+        assert(c);
+        assert(ret);
+
+        if (!c->numa_policy.nodes.set) {
+                log_debug("Can't derive CPU affinity mask from NUMA mask because NUMA mask is not set, ignoring");
+                return 0;
+        }
+
+        r = numa_to_cpu_set(&c->numa_policy, &s);
+        if (r < 0)
+                return r;
+
+        cpu_set_reset(ret);
```

Comment:
This is not necessary, right?

Reply:
I wanted to make sure that returned CPUSet contains only CPUs that correspond to NUMA mask hence I am resetting it before I add there list of CPUs derived from NUMA mask.

Reply:
Oh right, this is not about `s` but about `ret`. Never mind.